packages/language-server/src/plugins/typescript/features/CodeActionsProvider.ts

```diff
@@ -1275,101 +1258,119 @@ export class CodeActionsProviderImpl implements CodeActionsProvider {
-            const typeString = declaration
-                ? typeChecker.typeToString(typeChecker.getTypeOfSymbolAtLocation(p, declaration))
-                : '';
+        const edits = [document.scriptInfo, document.moduleScriptInfo]
+            .map((info) => {
+                if (!info) {
+                    return;
+                }
 
-            return { name: p.name, typeString };
-        });
+                const startTagNameEnd = document.positionAt(info.container.start + 7); // <script
+                const existingLangOffset = document
+                    .getText({
+                        start: startTagNameEnd,
+                        end: document.positionAt(info.start)
+                    })
+                    .indexOf('lang=');
+                if (existingLangOffset !== -1) {
+                    return {
+                        range: Range.create(startTagNameEnd, startTagNameEnd),
+                        newText: ' lang="ts"'
+                    };
+                }
+                return {
+                    range: Range.create(startTagNameEnd, startTagNameEnd),
+                    newText: ' lang="ts"'
+                };
+            })
+            .filter(isNotNullOrUndefined);
 
-        const returnType = typeChecker.typeToString(signature.getReturnType());
-        const useJsDoc =
-            tsDoc.scriptKind === ts.ScriptKind.JS || tsDoc.scriptKind === ts.ScriptKind.JSX;
-        const parametersText = (
-            useJsDoc
-                ? parameters.map((p) => p.name)
-                : parameters.map((p) => p.name + (p.typeString ? ': ' + p.typeString : ''))
-        ).join(', ');
-
-        const jsDoc = useJsDoc
-            ? ['/**', ...parameters.map((p) => ` * @param {${p.typeString}} ${p.name}`), ' */']
-            : [];
-
-        const newText = [
-            ...jsDoc,
-            `function ${identifier.text}(${parametersText})${
-                useJsDoc || returnType === 'any' ? '' : ': ' + returnType
-            } {`,
-            formatCodeBasis.indent +
-                `throw new Error(${quote}Function not implemented.${quote})` +
-                formatCodeBasis.semi,
-            '}'
-        ]
-            .map((line) => formatCodeBasis.baseIndent + line + formatCodeBasis.newLine)
-            .join('');
+        if (edits.length) {
+            return CodeAction.create(
+                'Add lang="ts" to <script> tag',
+                {
+                    documentChanges: [
+                        {
+                            textDocument: OptionalVersionedTextDocumentIdentifier.create(
+                                document.uri,
+                                null
+                            ),
+                            edits
+                        }
+                    ]
+                },
+                CodeActionKind.QuickFix
+            );
+        }
 
-        return [
+        return CodeAction.create(
```

Comment:
edits.length could also be `0` when we have a (module)scriptInfo but they did not produce code actions. I think we need to check for that and return nothing if that's the case